src/sentry/static/sentry/app/views/organizationIntegrations/index.jsx

```diff
@@ -87,10 +101,15 @@ export default class OrganizationIntegrations extends AsyncComponent {
     window.open(url, '_blank');
   };
 
+  handleAddIntegration(provider, props) {
```

Comment:
I really dislike how we've exported the add integration dialog, and would much rather we use the `AddIntegrationButton` and just have this thing open the modal. Of course, we can't just slap the button in there.

What I'm thinking is that we refactor the `AddIntegrationButton` into just `AddIntegration` and have it children be a render prop that passes the onCLick and then we can just stick a button in there.

```jsx
<AddIntegration ...>
  {onClick => <Button onClick={onClick}>Install</Button>}
</AddIntegration>

// Then for the alert warning

<AddIntegration ...>
  {onClick => <AlertLink onClick={handler} href="#"> ... </AlertLink>}
</AddIntegration>
```

Then we don't need this handleAddIntegration stuff here

Reply:
I actually also don't think this actually handles receiving the completed data after finishing the integration setup?